lib/libconfigcpp.c++

```diff
@@ -1106,7 +1106,7 @@ void Setting::remove(unsigned int idx)
 
 // ---------------------------------------------------------------------------
 
-Setting & Setting::add(const char *name, Setting::Type type)
+Setting & Setting::add(const char *name, Setting::Type type, const char *comment)
```

Comment:
should have an overload that takes a const std::string &comment (a simple inline function in libconfig.h++)

doc/libconfig.texi needs to be updated with documentation for the new APIs.

Reply:
also a couple of unit tests would be nice